src/kernels/raw/launcher/kernelLauncher.unit.test.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+'use strict';
+
+import { assert } from 'chai';
+import * as sinon from 'sinon';
+import { KernelLauncher } from './kernelLauncher.node';
+import { IPlatformService } from '../../../platform/common/platform/types';
+import { IConfigurationService, IDisposable } from '../../../platform/common/types';
+import {
+    IProcessServiceFactory,
+    IPythonExecutionFactory,
+    IPythonExecutionService
+} from '../../../platform/common/process/types.node';
+import { IPythonExtensionChecker } from '../../../platform/api/types';
+import { KernelEnvironmentVariablesService } from './kernelEnvVarsService.node';
+import { JupyterPaths } from '../finder/jupyterPaths.node';
+import { PythonKernelInterruptDaemon } from '../finder/pythonKernelInterruptDaemon.node';
+import { disposeAllDisposables } from '../../../platform/common/helpers';
+import { anything, instance, mock, when } from 'ts-mockito';
+import { IFileSystemNode } from '../../../platform/common/platform/types.node';
+import { PythonKernelConnectionMetadata } from '../../types';
+import { CancellationTokenSource, Disposable, EventEmitter, PortAutoForwardAction, Uri } from 'vscode';
+import { KernelProcess } from './kernelProcess.node';
+import { PortAttributesProviders } from '../port/portAttributeProvider.node';
+
+suite('kernel Launcher', () => {
```

Comment:
Converted an integration test into unit test
Plan is to do this more often where possible